src/classes/Player.js

```diff
@@ -132,6 +124,44 @@ class Player {
     }
   }
 
+  async createStream() {
+    const url = queue[this.songEntry];
+    if (url.includes('youtube.com')) {
+      return this.createYoutubeStream()
+
+    } else if (url.includes('soundcloud.com') && !!this.soundcloudClientID) {
+      return await this.createSoundcloudStream();
+    }
+  }
+
+  createYoutubeStream() {
+    const stream = ytdl(queue[this.songEntry], {
+      quality: 'highestaudio',
+      highWaterMark: 1 << 25
+    });
+
+    stream.once(streamEvents.info, ({ title }) => {
+      this.song = title;
+      if (!this.updateDispatcherStatus()) {
+        this.updateSongPresence();
+      }
+    });
+
+    return stream
+  }
+
+  async createSoundcloudStream() {
+    const stream = await scdl.download(queue[this.songEntry], this.soundcloudClientID);
+    const info = await scdl.getInfo(url, this.soundcloudClientID);
```

Comment:
Last change I promise! `url` here is not defined so it is throwing an error when trying to start the bot because of this.